src/seedu/addressbook/AddressBook.java

```diff
@@ -303,16 +302,43 @@ private static void setupDefaultFileForStorage() {
         storageFilePath = DEFAULT_STORAGE_FILEPATH;
         createFileIfMissing(storageFilePath);
     }
-
+    
     /**
-     * Returns true if the given file is acceptable.
-     * The file path is acceptable if it ends in '.txt'
-     * TODO: Implement a more rigorous validity checking.
+     * Returns true if the given file path is acceptable.
+     * The file path is acceptable if it has a folder that exists,
+     * has a file name that is acceptable to the operating system,
+     * File name must also contain an extentsion
      */
     private static boolean isValidFilePath(String filePath) {
-        return filePath.endsWith(".txt");
+        if (filePath == null) {
+            return false;
+        }
+        File filePathToValidate = new File(filePath);
+        return hasValidFileName(filePathToValidate) && hasValidFolder(filePathToValidate);
     }
-
+    
+    /**
+     * Returns true if the file path has a folder that exists
+     */
+    private static boolean hasValidFolder(File filePath) {
+        File folderToValidate = filePath.getAbsoluteFile().getParentFile();
+        return folderToValidate.exists();
+    }
+    
+    /**
+     * Returns true if file path has a valid file name.
+     * File name is valid if it has an extension and no reserved characters.
+     */
+    private static boolean hasValidFileName(File filePath) {
+        try {
+            File fileNameToValidate = new File(filePath.getName()).getCanonicalFile();
```

Comment:
Okay, after thinking about this, I'm still against this. Using `getCanonicalFile()` just to check if a path contains invalid characters or not is not using the method for its intended purpose, but just abusing one of its side effects. This is akin to writing a file out just to check if there are invalid characters or not.

Furthermore, `getCanonicalFile()` doesn't actually catch all invalid characters, and its is understandable why it is so -- it's contract does not promise that it would properly check for invalid characters or not.

E.g. on Windows:
```java
new File(">").getCanonicalFile();
```
will *not* throw an IOException, while
```java
Paths.get(">");
```
will throw an InvalidPathException correctly.

On the other hand, `Paths.get()` does make this guarantee. From the [InvalidPathException](https://docs.oracle.com/javase/7/docs/api/java/nio/file/InvalidPathException.html) docs:
> Unchecked exception thrown when path string cannot be converted into a Path because the path string contains invalid characters, or the path string is invalid for other file system specific reasons.

Of course, the underlying API may have bugs. However, whether the developers of that API actually fixes those bugs or not depends on whether we actually use their APIs as intended in the first place :-)